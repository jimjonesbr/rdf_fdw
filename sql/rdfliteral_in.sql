
SELECT '"foo"'::rdf_literal;
SELECT '"foo"^^xsd:string'::rdf_literal;
SELECT '"foo"@es'::rdf_literal;
SELECT '"foo"@es'::rdf_literal::text;
SELECT '"foo"@es'::rdf_literal::text::rdf_literal;
SELECT '"foo"^^xsd:string'::rdf_literal::text::rdf_literal;
SELECT '"nan"^^xsd:double'::rdf_literal;
SELECT '"NAN"^^xsd:double'::rdf_literal;
SELECT '"nAn"^^xsd:double'::rdf_literal;
SELECT '"forty-two"^^xsd:int'::rdf_literal;
SELECT '"invalid"^^xsd:dateTime'::rdf_literal;
SELECT '"25:00:00"^^xsd:time'::rdf_literal;
SELECT '"2025-13-01"^^xsd:date'::rdf_literal;
SELECT '"abc"^^invalid:datatype'::rdf_literal;
SELECT '"abc"@invalid_lang'::rdf_literal;
SELECT '"foo'::rdf_literal;
SELECT 'f"o"o'::rdf_literal;
SELECT 'ùÑû'::rdf_literal;
SELECT ''::rdf_literal;
SELECT '"'::rdf_literal; 
SELECT '"\""'::rdf_literal;
SELECT 'üòÄ'::rdf_literal;
SELECT '"x^^y"'::rdf_literal;           -- ‚Üí "x^^y"
SELECT '"a\\"b"@en'::rdf_literal;       -- ‚Üí "a\"b"@en
SELECT '"ùÑû"^^<http://example.org/dt>'::rdf_literal;


SELECT '"9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric;
SELECT '"-9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric;
SELECT '"9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric::rdf_literal;
SELECT '"-9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric::rdf_literal;

SELECT '"42.73"^^xsd:double'::rdf_literal::double precision;
SELECT '"-42.73"^^xsd:double'::rdf_literal::double precision;
SELECT '"42.73"^^xsd:double'::rdf_literal::double precision::rdf_literal;
SELECT '"-42.73"^^xsd:double'::rdf_literal::double precision::rdf_literal;
SELECT '"4.2E1"^^xsd:double'::rdf_literal::double precision;
SELECT '"-4.2E1"^^xsd:double'::rdf_literal::double precision;

SELECT 42.73::real::rdf_literal;
SELECT 42.73::real::rdf_literal::real;
SELECT (-42.73)::real::rdf_literal;
SELECT (-42.73)::real::rdf_literal::real;
SELECT 'INF'::real::rdf_literal;
SELECT 'INF'::real::rdf_literal::real;
SELECT '-INF'::real::rdf_literal;
SELECT '-INF'::real::rdf_literal::real;
SELECT 'NaN'::real::rdf_literal;
SELECT 'NaN'::real::rdf_literal::real;

SELECT 42::bigint::rdf_literal;
SELECT 42::bigint::rdf_literal::bigint;
SELECT (-42)::bigint::rdf_literal;
SELECT (-42)::bigint::rdf_literal::bigint;
SELECT 42746357267238767::bigint::rdf_literal;
SELECT 42746357267238767::bigint::rdf_literal::bigint;
SELECT (-42746357267238767)::bigint::rdf_literal;
SELECT (-42746357267238767)::bigint::rdf_literal::bigint;

SELECT 42::int::rdf_literal;
SELECT 42::int::rdf_literal::int;
SELECT (-42)::int::rdf_literal;
SELECT (-42)::int::rdf_literal::int;
SELECT 427463::int::rdf_literal;
SELECT 427463::int::rdf_literal::int;
SELECT (-427463)::int::rdf_literal;
SELECT (-427463)::int::rdf_literal::int;

SELECT 42::smallint::rdf_literal;
SELECT 42::smallint::rdf_literal::smallint;
SELECT (-42)::smallint::rdf_literal;
SELECT (-42)::smallint::rdf_literal::smallint;
SELECT 4273::smallint::rdf_literal;
SELECT 4273::smallint::rdf_literal::smallint;
SELECT (-4273)::smallint::rdf_literal;
SELECT (-4273)::smallint::rdf_literal::smallint;

SELECT '2025-04-25 18:44:38.149101+00'::timestamptz::rdf_literal;
SELECT '2025-04-25 18:44:38.149101+00'::timestamptz::rdf_literal::timestamptz;
SELECT '2025-04-25 18:44:38'::timestamptz::rdf_literal;
SELECT '2025-04-25 18:44:38'::timestamptz::rdf_literal::timestamptz;

SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal;
SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal::timestamp;
SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal;
SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal::timestamp;

SELECT '2020-05-12'::date::rdf_literal;
SELECT '2020-05-12'::date::rdf_literal::date;

SELECT '18:44:38'::time::rdf_literal;
SELECT '18:44:38'::time::rdf_literal::time;
SELECT '00:00:00'::time::rdf_literal;
SELECT '00:00:00'::time::rdf_literal::time;

SELECT '04:05:06-08:00'::timetz::rdf_literal;
SELECT '04:05:06-08:00'::timetz::rdf_literal::timetz;
SELECT '04:05:06 PST'::timetz::rdf_literal;
SELECT '04:05:06 PST'::timetz::rdf_literal::timetz;

SELECT true::rdf_literal;
SELECT false::rdf_literal;
SELECT true::rdf_literal::boolean;
SELECT false::rdf_literal::boolean;
SELECT (1=1)::rdf_literal::boolean;
SELECT (1<>1)::rdf_literal::boolean;

SELECT '1 day'::interval::rdf_literal;
SELECT '1 hour 30 minutes'::interval::rdf_literal;
SELECT '2 years 3 months'::interval::rdf_literal;
SELECT '5 days 12 hours'::interval::rdf_literal;
SELECT '1 year 2 months 3 days 4 hours 5 minutes 6 seconds'::interval::rdf_literal;
SELECT '5.123456 seconds'::interval::rdf_literal;
SELECT '0.000001 seconds'::interval::rdf_literal;
SELECT '1 minute 0.5 seconds'::interval::rdf_literal;
SELECT '-1 year -2 months'::interval::rdf_literal;
SELECT '-3 days -4 hours'::interval::rdf_literal;
SELECT '0 seconds'::interval::rdf_literal;
SELECT '1 year 2 months 3 days 4 hours 5 minutes 6 seconds'::interval::rdf_literal::interval;
SELECT '-1 year -2 months'::interval::rdf_literal::interval;
SELECT '5.123456 seconds'::interval::rdf_literal::interval;
SELECT '0 seconds'::interval::rdf_literal::interval;
SELECT '0.000001 seconds'::interval::rdf_literal::interval;