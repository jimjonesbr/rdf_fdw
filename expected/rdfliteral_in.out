SELECT '"foo"'::rdf_literal;
 rdf_literal 
-------------
 "foo"
(1 row)

SELECT '"foo"^^xsd:string'::rdf_literal;
                   rdf_literal                    
--------------------------------------------------
 "foo"^^<http://www.w3.org/2001/XMLSchema#string>
(1 row)

SELECT '"foo"@es'::rdf_literal;
 rdf_literal 
-------------
 "foo"@es
(1 row)

SELECT '"foo"@es'::rdf_literal::text;
   text   
----------
 "foo"@es
(1 row)

SELECT '"foo"@es'::rdf_literal::text::rdf_literal;
 rdf_literal 
-------------
 "foo"@es
(1 row)

SELECT '"foo"^^xsd:string'::rdf_literal::text::rdf_literal;
                   rdf_literal                    
--------------------------------------------------
 "foo"^^<http://www.w3.org/2001/XMLSchema#string>
(1 row)

SELECT '"nan"^^xsd:double'::rdf_literal;
                   rdf_literal                    
--------------------------------------------------
 "nan"^^<http://www.w3.org/2001/XMLSchema#double>
(1 row)

SELECT '"NAN"^^xsd:double'::rdf_literal;
                   rdf_literal                    
--------------------------------------------------
 "NAN"^^<http://www.w3.org/2001/XMLSchema#double>
(1 row)

SELECT '"nAn"^^xsd:double'::rdf_literal;
                   rdf_literal                    
--------------------------------------------------
 "nAn"^^<http://www.w3.org/2001/XMLSchema#double>
(1 row)

SELECT '"forty-two"^^xsd:int'::rdf_literal;
ERROR:  invalid lexical form for xsd:int: "forty-two"
LINE 1: SELECT '"forty-two"^^xsd:int'::rdf_literal;
               ^
SELECT '"invalid"^^xsd:dateTime'::rdf_literal;
ERROR:  invalid lexical form for xsd:dateTime: "invalid"
LINE 1: SELECT '"invalid"^^xsd:dateTime'::rdf_literal;
               ^
SELECT '"25:00:00"^^xsd:time'::rdf_literal;
ERROR:  invalid lexical form for xsd:time: "25:00:00"
LINE 1: SELECT '"25:00:00"^^xsd:time'::rdf_literal;
               ^
SELECT '"2025-13-01"^^xsd:date'::rdf_literal;
ERROR:  invalid lexical form for xsd:date: "2025-13-01"
LINE 1: SELECT '"2025-13-01"^^xsd:date'::rdf_literal;
               ^
SELECT '"abc"^^invalid:datatype'::rdf_literal;
        rdf_literal        
---------------------------
 "abc"^^<invalid:datatype>
(1 row)

SELECT '"abc"@invalid_lang'::rdf_literal;
ERROR:  invalid language tag: "invalid_lang"
LINE 1: SELECT '"abc"@invalid_lang'::rdf_literal;
               ^
SELECT '"foo'::rdf_literal;
 rdf_literal 
-------------
 "\"foo"
(1 row)

SELECT 'f"o"o'::rdf_literal;
 rdf_literal 
-------------
 "f"o"o"
(1 row)

SELECT 'ùÑû'::rdf_literal;
 rdf_literal 
-------------
 "ùÑû"
(1 row)

SELECT ''::rdf_literal;
 rdf_literal 
-------------
 ""
(1 row)

SELECT '"'::rdf_literal; 
 rdf_literal 
-------------
 "\""
(1 row)

SELECT '"\""'::rdf_literal;
 rdf_literal 
-------------
 "\""
(1 row)

SELECT 'üòÄ'::rdf_literal;
 rdf_literal 
-------------
 "üòÄ"
(1 row)

SELECT '"x^^y"'::rdf_literal;           -- ‚Üí "x^^y"
 rdf_literal 
-------------
 "x^^y"
(1 row)

SELECT '"a\\"b"@en'::rdf_literal;       -- ‚Üí "a\"b"@en
 rdf_literal 
-------------
 "a\\"b"@en
(1 row)

SELECT '"ùÑû"^^<http://example.org/dt>'::rdf_literal;
         rdf_literal          
------------------------------
 "ùÑû"^^<http://example.org/dt>
(1 row)

SELECT '"9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric;
       numeric        
----------------------
 9223372036854775.807
(1 row)

SELECT '"-9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric;
        numeric        
-----------------------
 -9223372036854775.807
(1 row)

SELECT '"9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric::rdf_literal;
                            rdf_literal                             
--------------------------------------------------------------------
 "9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>
(1 row)

SELECT '"-9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>'::rdf_literal::numeric::rdf_literal;
                             rdf_literal                             
---------------------------------------------------------------------
 "-9223372036854775.807"^^<http://www.w3.org/2001/XMLSchema#decimal>
(1 row)

SELECT '"42.73"^^xsd:double'::rdf_literal::double precision;
 float8 
--------
  42.73
(1 row)

SELECT '"-42.73"^^xsd:double'::rdf_literal::double precision;
 float8 
--------
 -42.73
(1 row)

SELECT '"42.73"^^xsd:double'::rdf_literal::double precision::rdf_literal;
                    rdf_literal                     
----------------------------------------------------
 "42.73"^^<http://www.w3.org/2001/XMLSchema#double>
(1 row)

SELECT '"-42.73"^^xsd:double'::rdf_literal::double precision::rdf_literal;
                     rdf_literal                     
-----------------------------------------------------
 "-42.73"^^<http://www.w3.org/2001/XMLSchema#double>
(1 row)

SELECT '"4.2E1"^^xsd:double'::rdf_literal::double precision;
 float8 
--------
     42
(1 row)

SELECT '"-4.2E1"^^xsd:double'::rdf_literal::double precision;
 float8 
--------
    -42
(1 row)

SELECT 42.73::real::rdf_literal;
                    rdf_literal                    
---------------------------------------------------
 "42.73"^^<http://www.w3.org/2001/XMLSchema#float>
(1 row)

SELECT 42.73::real::rdf_literal::real;
 float4 
--------
  42.73
(1 row)

SELECT (-42.73)::real::rdf_literal;
                    rdf_literal                     
----------------------------------------------------
 "-42.73"^^<http://www.w3.org/2001/XMLSchema#float>
(1 row)

SELECT (-42.73)::real::rdf_literal::real;
 float4 
--------
 -42.73
(1 row)

SELECT 'INF'::real::rdf_literal;
                     rdf_literal                      
------------------------------------------------------
 "Infinity"^^<http://www.w3.org/2001/XMLSchema#float>
(1 row)

SELECT 'INF'::real::rdf_literal::real;
  float4  
----------
 Infinity
(1 row)

SELECT '-INF'::real::rdf_literal;
                      rdf_literal                      
-------------------------------------------------------
 "-Infinity"^^<http://www.w3.org/2001/XMLSchema#float>
(1 row)

SELECT '-INF'::real::rdf_literal::real;
  float4   
-----------
 -Infinity
(1 row)

SELECT 'NaN'::real::rdf_literal;
                   rdf_literal                   
-------------------------------------------------
 "NaN"^^<http://www.w3.org/2001/XMLSchema#float>
(1 row)

SELECT 'NaN'::real::rdf_literal::real;
 float4 
--------
    NaN
(1 row)

SELECT 42::bigint::rdf_literal;
                  rdf_literal                  
-----------------------------------------------
 "42"^^<http://www.w3.org/2001/XMLSchema#long>
(1 row)

SELECT 42::bigint::rdf_literal::bigint;
 int8 
------
   42
(1 row)

SELECT (-42)::bigint::rdf_literal;
                  rdf_literal                   
------------------------------------------------
 "-42"^^<http://www.w3.org/2001/XMLSchema#long>
(1 row)

SELECT (-42)::bigint::rdf_literal::bigint;
 int8 
------
  -42
(1 row)

SELECT 42746357267238767::bigint::rdf_literal;
                         rdf_literal                          
--------------------------------------------------------------
 "42746357267238767"^^<http://www.w3.org/2001/XMLSchema#long>
(1 row)

SELECT 42746357267238767::bigint::rdf_literal::bigint;
       int8        
-------------------
 42746357267238767
(1 row)

SELECT (-42746357267238767)::bigint::rdf_literal;
                          rdf_literal                          
---------------------------------------------------------------
 "-42746357267238767"^^<http://www.w3.org/2001/XMLSchema#long>
(1 row)

SELECT (-42746357267238767)::bigint::rdf_literal::bigint;
        int8        
--------------------
 -42746357267238767
(1 row)

SELECT 42::int::rdf_literal;
                 rdf_literal                  
----------------------------------------------
 "42"^^<http://www.w3.org/2001/XMLSchema#int>
(1 row)

SELECT 42::int::rdf_literal::int;
 int4 
------
   42
(1 row)

SELECT (-42)::int::rdf_literal;
                  rdf_literal                  
-----------------------------------------------
 "-42"^^<http://www.w3.org/2001/XMLSchema#int>
(1 row)

SELECT (-42)::int::rdf_literal::int;
 int4 
------
  -42
(1 row)

SELECT 427463::int::rdf_literal;
                   rdf_literal                    
--------------------------------------------------
 "427463"^^<http://www.w3.org/2001/XMLSchema#int>
(1 row)

SELECT 427463::int::rdf_literal::int;
  int4  
--------
 427463
(1 row)

SELECT (-427463)::int::rdf_literal;
                    rdf_literal                    
---------------------------------------------------
 "-427463"^^<http://www.w3.org/2001/XMLSchema#int>
(1 row)

SELECT (-427463)::int::rdf_literal::int;
  int4   
---------
 -427463
(1 row)

SELECT 42::smallint::rdf_literal;
                  rdf_literal                   
------------------------------------------------
 "42"^^<http://www.w3.org/2001/XMLSchema#short>
(1 row)

SELECT 42::smallint::rdf_literal::smallint;
 int2 
------
   42
(1 row)

SELECT (-42)::smallint::rdf_literal;
                   rdf_literal                   
-------------------------------------------------
 "-42"^^<http://www.w3.org/2001/XMLSchema#short>
(1 row)

SELECT (-42)::smallint::rdf_literal::smallint;
 int2 
------
  -42
(1 row)

SELECT 4273::smallint::rdf_literal;
                   rdf_literal                    
--------------------------------------------------
 "4273"^^<http://www.w3.org/2001/XMLSchema#short>
(1 row)

SELECT 4273::smallint::rdf_literal::smallint;
 int2 
------
 4273
(1 row)

SELECT (-4273)::smallint::rdf_literal;
                    rdf_literal                    
---------------------------------------------------
 "-4273"^^<http://www.w3.org/2001/XMLSchema#short>
(1 row)

SELECT (-4273)::smallint::rdf_literal::smallint;
 int2  
-------
 -4273
(1 row)

SELECT '2025-04-25 18:44:38.149101+00'::timestamptz::rdf_literal;
                                rdf_literal                                 
----------------------------------------------------------------------------
 "2025-04-25T18:44:38.149101Z"^^<http://www.w3.org/2001/XMLSchema#dateTime>
(1 row)

SELECT '2025-04-25 18:44:38.149101+00'::timestamptz::rdf_literal::timestamptz;
             timestamptz             
-------------------------------------
 Fri Apr 25 11:44:38.149101 2025 PDT
(1 row)

SELECT '2025-04-25 18:44:38'::timestamptz::rdf_literal;
                                rdf_literal                                 
----------------------------------------------------------------------------
 "2025-04-26T01:44:38.000000Z"^^<http://www.w3.org/2001/XMLSchema#dateTime>
(1 row)

SELECT '2025-04-25 18:44:38'::timestamptz::rdf_literal::timestamptz;
         timestamptz          
------------------------------
 Fri Apr 25 18:44:38 2025 PDT
(1 row)

SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal;
                            rdf_literal                             
--------------------------------------------------------------------
 "2025-04-25T18:44:38"^^<http://www.w3.org/2001/XMLSchema#dateTime>
(1 row)

SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal::timestamp;
        timestamp         
--------------------------
 Fri Apr 25 18:44:38 2025
(1 row)

SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal;
                            rdf_literal                             
--------------------------------------------------------------------
 "2025-04-25T18:44:38"^^<http://www.w3.org/2001/XMLSchema#dateTime>
(1 row)

SELECT '2025-04-25 18:44:38'::timestamp::rdf_literal::timestamp;
        timestamp         
--------------------------
 Fri Apr 25 18:44:38 2025
(1 row)

SELECT '2020-05-12'::date::rdf_literal;
                      rdf_literal                      
-------------------------------------------------------
 "2020-05-12"^^<http://www.w3.org/2001/XMLSchema#date>
(1 row)

SELECT '2020-05-12'::date::rdf_literal::date;
    date    
------------
 05-12-2020
(1 row)

SELECT '18:44:38'::time::rdf_literal;
                     rdf_literal                     
-----------------------------------------------------
 "18:44:38"^^<http://www.w3.org/2001/XMLSchema#time>
(1 row)

SELECT '18:44:38'::time::rdf_literal::time;
   time   
----------
 18:44:38
(1 row)

SELECT '00:00:00'::time::rdf_literal;
                     rdf_literal                     
-----------------------------------------------------
 "00:00:00"^^<http://www.w3.org/2001/XMLSchema#time>
(1 row)

SELECT '00:00:00'::time::rdf_literal::time;
   time   
----------
 00:00:00
(1 row)

SELECT '04:05:06-08:00'::timetz::rdf_literal;
                      rdf_literal                       
--------------------------------------------------------
 "04:05:06-08"^^<http://www.w3.org/2001/XMLSchema#time>
(1 row)

SELECT '04:05:06-08:00'::timetz::rdf_literal::timetz;
   timetz    
-------------
 04:05:06-08
(1 row)

SELECT '04:05:06 PST'::timetz::rdf_literal;
                      rdf_literal                       
--------------------------------------------------------
 "04:05:06-08"^^<http://www.w3.org/2001/XMLSchema#time>
(1 row)

SELECT '04:05:06 PST'::timetz::rdf_literal::timetz;
   timetz    
-------------
 04:05:06-08
(1 row)

SELECT true::rdf_literal;
                    rdf_literal                     
----------------------------------------------------
 "true"^^<http://www.w3.org/2001/XMLSchema#boolean>
(1 row)

SELECT false::rdf_literal;
                     rdf_literal                     
-----------------------------------------------------
 "false"^^<http://www.w3.org/2001/XMLSchema#boolean>
(1 row)

SELECT true::rdf_literal::boolean;
 bool 
------
 t
(1 row)

SELECT false::rdf_literal::boolean;
 bool 
------
 f
(1 row)

SELECT (1=1)::rdf_literal::boolean;
 bool 
------
 t
(1 row)

SELECT (1<>1)::rdf_literal::boolean;
 bool 
------
 f
(1 row)

SELECT '1 day'::interval::rdf_literal;
                    rdf_literal                     
----------------------------------------------------
 "P1D"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '1 hour 30 minutes'::interval::rdf_literal;
                      rdf_literal                       
--------------------------------------------------------
 "PT1H30M"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '2 years 3 months'::interval::rdf_literal;
                     rdf_literal                      
------------------------------------------------------
 "P2Y3M"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '5 days 12 hours'::interval::rdf_literal;
                      rdf_literal                       
--------------------------------------------------------
 "P5DT12H"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '1 year 2 months 3 days 4 hours 5 minutes 6 seconds'::interval::rdf_literal;
                          rdf_literal                          
---------------------------------------------------------------
 "P1Y2M3DT4H5M6S"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '5.123456 seconds'::interval::rdf_literal;
                        rdf_literal                         
------------------------------------------------------------
 "PT5.123456S"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '0.000001 seconds'::interval::rdf_literal;
                        rdf_literal                         
------------------------------------------------------------
 "PT0.000001S"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '1 minute 0.5 seconds'::interval::rdf_literal;
                         rdf_literal                          
--------------------------------------------------------------
 "PT1M0.500000S"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '-1 year -2 months'::interval::rdf_literal;
                      rdf_literal                      
-------------------------------------------------------
 "-P1Y2M"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '-3 days -4 hours'::interval::rdf_literal;
                      rdf_literal                       
--------------------------------------------------------
 "-P3DT4H"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '0 seconds'::interval::rdf_literal;
                     rdf_literal                     
-----------------------------------------------------
 "PT0S"^^<http://www.w3.org/2001/XMLSchema#duration>
(1 row)

SELECT '1 year 2 months 3 days 4 hours 5 minutes 6 seconds'::interval::rdf_literal::interval;
                   interval                   
----------------------------------------------
 @ 1 year 2 mons 3 days 4 hours 5 mins 6 secs
(1 row)

SELECT '-1 year -2 months'::interval::rdf_literal::interval;
ERROR:  invalid input syntax for type interval: "-P1Y2M"
SELECT '5.123456 seconds'::interval::rdf_literal::interval;
    interval     
-----------------
 @ 5.123456 secs
(1 row)

SELECT '0 seconds'::interval::rdf_literal::interval;
 interval 
----------
 @ 0
(1 row)

SELECT '0.000001 seconds'::interval::rdf_literal::interval;
    interval     
-----------------
 @ 0.000001 secs
(1 row)

